### PR TITLE
Allow to built release-candidate

### PR DESCRIPTION
**Pull request by Zelldon**
I added another event subprocess to allow to build a release candidate. This helps to not find all information by hand, but with that we can use our existing tooling and information we have in the process.

zeebe_release

### CODE REVIEW

**Comment by Zelldon** [comment]
i updated the PR to build per default an rc which can be used by other teams and the e2e tests.

\cc @deepthidevaki 


build-release-candidate
zeebe_release

File: src/main/resources/zeebe_release.bpmn [hunk 1]
```diff
@@ -281,97 +283,58 @@
...
     </bpmn:textAnnotation>
     <bpmn:association id="Association_0uafnuq" sourceRef="StartEvent_1" targetRef="TextAnnotation_15g9k3f" />
+    <bpmn:textAnnotation id="TextAnnotation_149oh6t">
+      <bpmn:text>Used by other teams to test against and by the e2e tests.</bpmn:text>
+    </bpmn:textAnnotation>
+    <bpmn:association id="Association_10w5l9m" sourceRef="Activity_175is1m" targetRef="TextAnnotation_149oh6t" />
   </bpmn:process>
+  <bpmn:message id="Message_36527rk" name="rc">
+    <bpmn:extensionElements>
+      <zeebe:subscription correlationKey="=release_branch_name" />
```
Comment (saig0) [inline src/main/resources/zeebe_release.bpmn:329]: :x: The variable `release_branch_name` is not defined when the process instance is created. As a result, an incident is created. 

The variable is created when completing the form of the task `Create release branch`.

File: src/main/resources/build-release-candidate.bpmn [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    <bpmn:startEvent id="StartEvent_1" name="Build release candidate">
+      <bpmn:outgoing>Flow_1yahb20</bpmn:outgoing>
+    </bpmn:startEvent>
+    <bpmn:userTask id="Activity_0wjyoge" name="Trigger release candidate">
+      <bpmn:extensionElements>
+        <zeebe:formDefinition formKey="camunda-forms:bpmn:UserTaskForm_38jg436" />
+      </bpmn:extensionElements>
+      <bpmn:incoming>Flow_1yahb20</bpmn:incoming>
+      <bpmn:outgoing>Flow_0laljwe</bpmn:outgoing>
+    </bpmn:userTask>
```
Comment (saig0) [inline src/main/resources/build-release-candidate.bpmn:51]: :wrench: We could adjust the form to make it more specific for building the release candidate. This is the same for the other user tasks in the process.

I guess that it uses the form `release_trigger_release.form`.

We could change the variable `release_version` to `release_version + "-rc1"`.

File: src/main/resources/build-release-candidate.bpmn [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      <bpmn:extensionElements>
+        <zeebe:taskDefinition type="io.camunda:slack:1" />
+        <zeebe:ioMapping>
+          <zeebe:input source="chat.postMessage" target="method" />
+          <zeebe:input source="secrets.SLACK_OAUTH_TOKEN" target="token" />
+          <zeebe:input source="#c8-release-announcements" target="data.channel" />
+          <zeebe:input source="= &#34;Hi :wave: team! There is a new Zeebe version &#34; + release_version + &#34; available that you can test against.&#34;" target="data.text" />
```
Comment (saig0) [inline src/main/resources/build-release-candidate.bpmn:66]: :x: We should adjust the Slack message. Currently, we would announce the release version when we just built the release candidate.

File: src/main/resources/build-release-candidate.bpmn [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+      <bpmn:incoming>Flow_0sjbp2b</bpmn:incoming>
+      <bpmn:outgoing>Flow_1g3i425</bpmn:outgoing>
+    </bpmn:serviceTask>
+    <bpmn:callActivity id="Activity_1aiu7sp" name="Release Zeebe process test">
+      <bpmn:extensionElements>
+        <zeebe:calledElement processId="zeebe-process-test-release" propagateAllChildVariables="false" />
+      </bpmn:extensionElements>
+      <bpmn:incoming>Flow_1g3i425</bpmn:incoming>
+      <bpmn:outgoing>Flow_0084v7j</bpmn:outgoing>
+    </bpmn:callActivity>
```
Comment (saig0) [inline src/main/resources/build-release-candidate.bpmn:82]: :x: Same here. We should adjust the variables. Currently, we would try to release Zeebe process test for the release version that is not released yet.

**Review by saig0 — CHANGES_REQUESTED** [review]
@Zelldon thank you for improving the release process. :+1: 

I played with the new process and found a few issues. Please have a look.
I'm happy to have a chat about the comments.

File: src/main/resources/zeebe_release.bpmn [hunk 5]
```diff
@@ -218,7 +220,7 @@
     <bpmn:sequenceFlow id="Flow_12eezq7" name="Yes" sourceRef="Gateway_1afx4hl" targetRef="Gateway_113spke">
       <bpmn:conditionExpression xsi:type="bpmn:tFormalExpression">= release_type = "major" or release_type = "minor"</bpmn:conditionExpression>
     </bpmn:sequenceFlow>
-    <bpmn:sequenceFlow id="Flow_1n61zbo" name="No" sourceRef="Gateway_1afx4hl" targetRef="Gateway_15n27wl" />
+    <bpmn:sequenceFlow id="Flow_1n61zbo" name="No - Alpha release" sourceRef="Gateway_1afx4hl" targetRef="Gateway_15n27wl" />
```
Comment (saig0) [inline src/main/resources/zeebe_release.bpmn:223]: :wrench: We should adjust the name. The path is taken for `alpha` and `patch` releases.

**Comment by Zelldon** [comment]
I will close this for now and defer to https://github.com/camunda/zeebe/issues/10651

**Pull request closed without merging**